packages/@lwc/babel-plugin-component/src/decorators/wire/validate.ts

```diff
@@ -105,6 +101,60 @@ function validateWireParameters(path: NodePath, state: LwcBabelPluginPass) {
             state
         );
     }
+
+    for (const prop of config.get('properties')) {
+        // Only validate {[computed]: true} object properties; {static: true} props are all valid
+        // and we ignore {...spreads} and {methods(){}}
+        if (!prop.isObjectProperty() || !prop.node.computed) continue;
+
+        const key: NodePath = prop.get('key');
+        if (key.isIdentifier()) {
+            // Only allow identifiers that originated from a `const` declaration
+            const binding = key.scope.getBinding(key.node.name);
+            // TODO [#3956]: Investigate allowing imported constants
+            if (binding?.kind === 'const') continue;
+            // By default, the identifier `undefined` has no binding (when it's actually undefined),
+            // but has a binding if it's used as a variable (e.g. `let undefined = "don't do this"`)
+            if (key.node.name === 'undefined' && !binding) continue;
+        } else if (key.isLiteral()) {
+            // A literal can be a regexp, template literal, or primitive; only allow primitives
+            if (key.isTemplateLiteral()) {
+                // A template literal is not guaranteed to always result in the same value
+                // (e.g. `${Math.random()}`), so we disallow them entirely.
+                // TODO [#3956]: Investigate allowing template literals
+                throw generateError(
+                    key,
+                    {
+                        errorInfo: DecoratorErrors.COMPUTED_PROPERTY_CANNOT_BE_TEMPLATE_LITERAL,
+                    },
+                    state
+                );
+            } else if (!key.isRegExpLiteral()) {
+                continue;
+            }
+        }
+
+        throw generateError(
+            key,
+            {
+                errorInfo: DecoratorErrors.COMPUTED_PROPERTY_MUST_BE_CONSTANT_OR_LITERAL,
+            },
+            state
+        );
+    }
+}
+
+function validateWireParameters(path: NodePath, state: LwcBabelPluginPass) {
+    const expressionArguments = path.get('expression.arguments');
+    if (Array.isArray(expressionArguments)) {
+        // Multiple arguments: should be [id, config?]
+        const [id, config] = expressionArguments;
+        validateWireId(id, path, state);
+        if (config) validateWireConfig(config, path, state);
+    } else {
+        // Single argument: should just be id
```

Comment:
As far as I can tell, this never actually occurs (the code prior to this change doesn't handle this case and would have thrown). I don't actually know whether it's true that a single argument would be the ID here, but it seems a reasonable guess.

Reply:
The [docs](https://developer.salesforce.com/docs/platform/lwc/guide/data-wire-service-about.html) are not clear, but it does seem that the `config` is optional. A quick search for `@wire` in SourceGraph reveals several instances of `@wire` being used with just the single argument. We should probably add a test for this.

Also you are correct, this line is not hit per code coverage.